memote/support/biomass.py

```diff
@@ -176,14 +180,56 @@ def find_direct_metabolites(model, reaction):
         that are taken up to be consumed by the biomass reaction only.
 
     """
-    transport_reactions = set(helpers.find_transport_reactions(model))
-    exchange_reactions = set(model.exchanges)
-    biomass_reactions = set(helpers.find_biomass_reaction(model))
-
-    combined_set = transport_reactions | exchange_reactions | biomass_reactions
+    biomass_rxns = set(helpers.find_biomass_reaction(model))
+    tra_bou_bio_rxns = helpers.find_tra_bou_bio_reactions(model, biomass_rxns)
     precursors = find_biomass_precursors(model, reaction)
+    tra_bou_bio_mets = [met for met in precursors if
+                        met.reactions.issubset(tra_bou_bio_rxns)]
+    rxns_of_interest = set([rxn for met in tra_bou_bio_mets
+                            for rxn in met.reactions
+                            if rxn not in biomass_rxns])
+
+    solution = model.optimize()
+    if solution.objective_value == 0:
+        raise ValueError("objective value is zero. Model will not grow.")
 
-    return [met for met in precursors if met.reactions.issubset(combined_set)]
+    tra_bou_bio_fluxes = solution.fluxes[get_ids(rxns_of_interest)]
+    flux_sum = pd.DataFrame(index=tra_bou_bio_mets, columns=["sum"], data=0).T
```

Comment:
I have a feeling that using a pandas data frame here is a bit of an overkill and a simple dictionary could be just as effective since you're essentially only needing the `tra_bou_bio_mets` as keys and the corresponding flux sums as values.